homeassistant/components/august.py

```diff
@@ -168,6 +178,22 @@ def __init__(self, api, access_token):
         self._door_state_by_id = {}
         self._activities_by_id = {}
 
+        @callback
+        def august_api_stop(event):
+            """Close the API HTTP session."""
+            _LOGGER.debug("Closing August HTTP session")
+
+            try:
+                self._api.http_session.close()
+                self._api.http_session = None
+            except RequestException:
+                pass
+            _LOGGER.debug("August HTTP session closed.")
+
+        self._hass.bus.listen_once(
+            EVENT_HOMEASSISTANT_STOP, august_api_stop)
+        _LOGGER.debug("Registered for HASS stop event")
```

Comment:
Please add this object a `stop` method and register this inside `setup`. It's a bad practice to do this inside a init

Reply:
@pvizeli Created PR #18582 for this, can you please check and confirm that this is indeed the way it should be. :-)

@balloob new PR #18582 created to address the comment.